app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if (user.role = 'admin')
+      can :manage, :all
+    else
+      can :read, :all
+      can :manage, Post, author_id: user.id
+      can :manage, Comment, user_id: user.id
+      can :destroy, Post, author_id: user.id, role: 'admin'
+    end
+  end
+end
```

Comment:
- [ ] Kindly go through your app. A user who is neither the author of the post or comment nor an admin can see the delete button on the views and delete posts and comments. Kindly fix this. Only the user who wrote the post/comment and an admin can see the buttons and can delete them.